src/modules/roc_core/target_musl/roc_core/backtrace.cpp

```diff
@@ -6,15 +6,218 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
+#ifndef LIBUNWIND_HH
+#define LIBUNWIND_HH
+#define UNW_LOCAL_ONLY
+#include <libunwind.h>
+#endif
+#include <cxxabi.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
 #include "roc_core/backtrace.h"
 
+/* Function to convert unsigned int to decimal/hex string.
+ * This function is signal safe.
+ * Parameters : 
+ * number = the number to be converted.
+ * base = 10 for decimal to char* and 16 for hex to char* conversion.
+ * output_string = the decimal/hex string
+ */
+void uint_to_string(unsigned int number, int base, char *output_string)
+{
+	int i = 12;
+	int j = 0;
+	do {
+		output_string[i] = "0123456789abcdef"[number % base];
+		i--;
+		number = number/base;
+	} while(number > 0);
+
+	while(++i < 13)
+		output_string[j++] = output_string[i];
+	output_string[j] = 0;
+}
+
+/* Return size of backtrace stack by unwinding it.
+ * This must be signal safe.
+ */
+ssize_t backtrace()
```

Comment:
As far as I can see, we need this function and performing the unwind twice only to check if the backtrace is available and to print "No backtrace available" if it's not. I think it would be an overkill. In the rare case when there's no backtrace it's acceptable to print just `Backtrace:` and then nothing. So I suggest to get rid of the `backtrace()` function.

Reply:
I was thinking of changing `backtrace()` to `is_backtrace_available()` by just checking if at-least 1 entry is available in the backtrace.  It can be implemented to return just a boolean value. 
true - backtrace available and false - no backtrace available.

Reply:
If you think we really need this check, `is_backtrace_available()` is a good option. In this case, can we also avoid iterating the complete backtrace in this function?